### PR TITLE
add TransportContext.GetTlsTokenBindings and TokenBinding class

### LINKED ISSUE #12136: Port System.Net.TransportContext missing members
```csharp
    public abstract class TransportContext {
        public virtual IEnumerable<TokenBinding> GetTlsTokenBindings();
    }
```

### PR DESCRIPTION
**Pull request by geoffkizer**
Fixes #12136

### CODE REVIEW

**Comment by geoffkizer** [comment]
@stephentoub @ericeil @CIPop @davidsh

**Comment by geoffkizer** [comment]
@dotnet-bot 
Innerloop Windows_NT Debug Build and Test
Innerloop Windows_NT Release Build and Test

**Comment by geoffkizer** [comment]
@dotnet-bot 
 Innerloop Windows_NT Debug Build and Test

**Comment by geoffkizer** [comment]
@dotnet-bot help

**Comment by geoffkizer** [comment]
@dotnet-bot test Innerloop Windows_NT Debug Build and Test

**Comment by geoffkizer** [comment]
@dotnet-bot test Innerloop Windows_NT Debug Build and Test

**Comment by geoffkizer** [comment]
@dotnet-bot test Innerloop Windows_NT Release Build and Test

**Comment by geoffkizer** [comment]
@dotnet-bot test Innerloop Windows_NT Debug Build and Test

**Comment by geoffkizer** [comment]
@dotnet-bot test Innerloop Windows_NT Release Build and Test

File: src/System.Net.Primitives/ref/System.Net.Primitives.cs [hunk 1]
```diff
@@ -471,4 +472,15 @@ public enum ChannelBindingKind
         Unique = 25,
         Unknown = 0,
     }
+    public sealed class TokenBinding
+    {
+        public TokenBinding(TokenBindingType bindingType, byte[] rawTokenBindingId) { }
```
Comment (stephentoub) [inline src/System.Net.Primitives/ref/System.Net.Primitives.cs]: Also, didn't we want it to be protected?

File: src/System.Net.Primitives/ref/System.Net.Primitives.cs [hunk 2]
```diff
@@ -471,4 +472,15 @@ public enum ChannelBindingKind
         Unique = 25,
         Unknown = 0,
     }
+    public sealed class TokenBinding
```
Comment (stephentoub) [inline src/System.Net.Primitives/ref/System.Net.Primitives.cs]: This is sealed?

File: src/System.Net.Primitives/src/System/Net/TransportContext.cs [hunk 3]
```diff
@@ -2,12 +2,18 @@
...
 
 namespace System.Net
 {
     public abstract class TransportContext
     {
         public abstract ChannelBinding GetChannelBinding(ChannelBindingKind kind);
+
+        public virtual IEnumerable<TokenBinding> GetTlsTokenBindings()
+        {
+            throw new NotSupportedException();
```
Comment (stephentoub) [inline src/System.Net.Primitives/src/System/Net/TransportContext.cs]: Maybe PlatformNotSupportedException?  @weshaggard, what's our guidance for a case like this on whether it's NSE or PNSE?

Comment (stephentoub) [inline src/System.Net.Primitives/src/System/Net/TransportContext.cs]: Oh, wait, nevermind, I see this is exactly what desktop throws.  NSE is good then.

File: src/System.Net.Primitives/src/System/Security/Authentication/ExtendedProtection/TokenBinding.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Security.Authentication.ExtendedProtection
+{
+    public sealed class TokenBinding
+    {
+        private TokenBindingType _bindingType;
+        private byte[] _rawTokenBindingId;
```
Comment (stephentoub) [inline src/System.Net.Primitives/src/System/Security/Authentication/ExtendedProtection/TokenBinding.cs]: readonly on both of these?

File: src/System.Net.Primitives/src/System/Security/Authentication/ExtendedProtection/TokenBinding.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+        public TokenBinding(TokenBindingType bindingType, byte[] rawTokenBindingId)
+        {
+            _bindingType = bindingType;
+            _rawTokenBindingId = rawTokenBindingId;
+        }
+
+        public byte[] GetRawTokenBindingId()
+        {
+            return (_rawTokenBindingId != null) ? (byte[])_rawTokenBindingId.Clone() : null;
```
Comment (stephentoub) [inline src/System.Net.Primitives/src/System/Security/Authentication/ExtendedProtection/TokenBinding.cs]: Consider:
```C#
public byte[] GetRawTokenBindingId() => (byte[])_rawTokenBindingId?.Clone();
```

File: src/System.Net.Primitives/ref/System.Net.Primitives.cs (hunk 2 above)
Comment (davidsh) [inline src/System.Net.Primitives/ref/System.Net.Primitives.cs]: it should not be sealed.

I don't think it is "sealed" on Desktop either.

File: src/System.Net.Primitives/ref/System.Net.Primitives.cs (hunk 1 above)
Comment (geoffkizer) [inline src/System.Net.Primitives/ref/System.Net.Primitives.cs]: @stephentoub I tried adding ifdefs etc as you described, but it still fails.  I have a feeling there's something I still haven't wired up.

The ref/System.Net.Primitives.builds didn't exist, so I created it based on the one you linked.  Do I have to point to this somewhere else in order to make it all work?

**Comment by geoffkizer** [comment]
Cut from netstandard 2.0.  See #12136.

**Pull request closed without merging**